src/platform/DeviceControlServer.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides the implementation of the DeviceControlServer object.
+ */
+
+#include <platform/internal/DeviceControlServer.h>
+
+#include <platform/ConfigurationManager.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+// Conforms to ISO 3166-1 Alpha-2
+static constexpr size_t kCountryCodeLen = 2;
```

Comment:
How about making the caller either send null terminated string (tiny string, cost is small) and validate this below or have the caller pass in the length.

Reply:
Acked